### ISSUE TITLE
Enhancement: [no-namespace] allow-type-only

### ISSUE THREAD

**Issue by Jack-Works** [typescript-eslint/typescript-eslint#6572]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-namespace/

### Description

the namespace is helpful as a container for types, I hope the rule can allow a pure-type namespace but bans the rest.

The current option `allowDeclarations` also does not apply.

### Fail

```typescript
namespace T {
  export enum T1 { val }
}
const x = T1.val
```

### Pass

```typescript
namespace T {
  export interface T1 {}
}

const x: T.T1 = {}
```

**Comment by bradzacher** [comment]
> the namespace is helpful as a container for types

I'm not sure I see the benefit of barreling up types within a namespace in the general case.
If anything it can be counter-productive because it means consumers *have* to use your namespace in order to consume a type, instead of being able to selectively import a name.

---

Given that this rule has existed in its current form essentially since the beginning of the project without this option - I don't think there are many people in the community that are looking for such an option.
For this reason we are going to reject this proposal from this project.
If this is important to your codebase, feel free to fork our rule to suite your style!
